lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -294,11 +295,19 @@ class FontSize extends Gatherer {
       .sort((a, b) => b.textLength - a.textLength)
       .slice(0, MAX_NODES_SOURCE_RULE_FETCHED)
       .map(async failingNode => {
-        failingNode.cssRule = await fetchSourceRule(driver, failingNode.node);
+        try {
+          const cssRule = await fetchSourceRule(driver, failingNode.node);
+          failingNode.cssRule = cssRule;
+        } catch (err) {
+          // The node was deleted. Don't set `cssRule` in that case.
+          log.error('font-size', err.message);
+        }
         return failingNode;
       });
 
-    const analyzedFailingNodesData = await Promise.all(analysisPromises);
+    const analyzedFailingNodesData = (await Promise.all(analysisPromises))
+      // Throw out the nodes that got deleted.
```

Comment:
(also when `fetchSourceRule` returns undefined)

Reply:
does it?

![image](https://user-images.githubusercontent.com/4071474/68343840-94d65d80-00a2-11ea-8c0e-ba704e4e0655.png)

Reply:
Oh I see. Good catch, but this distinction is definitely going to be overlooked like I just did the next time someone is doing something in here (at least without a test for it, which is going to be annoying/hard).

Should we just treat these two failures the same to keep it simpler (e.g. count them both in the total length since the elements were at least in the DOM at some point)? Or rearrange the mapping/filtering to make it clearer.

Reply:
I think we shouldn't keep the nodes we know were _deleted_, because in the case of a timer (that can be modified many many many times) the percentages will be thrown off what is visible at any one point.

I added some comments to make it more clear. Importantly, that second filter isn't filtering result _out_ of the artifact, it is just assigning the stylesheet object to the nodes that we could attribute a stylesheet to.

Reply:
> I think we shouldn't keep the nodes we know were deleted, because in the case of a timer (that can be modified many many many times) the percentages will be thrown off what is visible at any one point.

Wouldn't it be more correct to include it, though? The gatherer is using `DOM.getFlattenedDocument` to get a snapshot of the DOM at a single point in time, so the percentage including it would be accurate for that moment (which is when all the text lengths come from, not from when we finally get to calling `CSS.getMatchedStylesForNode`)

e.g. if the timer is the only too-small text on screen, dropping it would give an `analyzedFailingTextLength` of 0 and no node listing when we do know the node that was failing, and in theory have a decent selector to that would indicate the issue was the timer element.

Reply:
hmmm this sounds compelling. I was thinking we'd somehow have many copies of the "same" node, but that doesn't make sense b/c we just take a snapshot.